### PR TITLE
Make station size limit optional

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
This makes the station length check optional. In vanilla, the effective length limit is 16 tiles, implemented as a check for 8 tiles from the station centre. I've implemented the function performing the check, and added an option to bypass the check.

Screenshot (1)

### CODE REVIEW

File: src/OpenLoco/src/GameCommands/Track/CreateTrainStation.cpp [hunk 1]
```diff
@@ -97,21 +87,6 @@ namespace OpenLoco::GameCommands
             }
             return std::make_pair(NearbyStationValidation::requiresNewStation, StationId::null);
         }
-        else
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/Track/CreateTrainStation.cpp:100]: Remove altogether rather than behind a cheat?

Comment (AaronVanGeffen) [reply src/OpenLoco/src/GameCommands/Track/CreateTrainStation.cpp:100]: Personally, I'd be happy to get rid of the limit entirely. I think 8 tiles is just rather short. Alternatively, perhaps we could double the limit to -say- 16 tiles, along with a cheat to remove the limit entirely?

File: src/OpenLoco/src/GameCommands/Track/CreateTrainStation.cpp [hunk 2]
```diff
@@ -42,14 +43,18 @@ namespace OpenLoco::GameCommands
...
     {
-        registers regs;
-        regs.ax = pos.x;
-        regs.cx = pos.y;
-        regs.dx = pos.z;
-        regs.ebx = enumValue(id);
-        return call(0x0048FEF4, regs) & X86_FLAG_CARRY;
+        const auto xDist = std::abs((pos.x + 16) - station->x);
+        const auto yDist = std::abs((pos.y + 16) - station->y);
+        const auto radius = std::max(xDist, yDist);
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/Track/CreateTrainStation.cpp:51]: there is a Math::manhattenDistance for this.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 24.02+ (???)
 ------------------------------------------------------------------------
 - Fix: [#2310] Newly created stations and the tracks they are on have zero clearance.
+- Change: [#126] Station lengths are no longer limited to 8 tiles (though still max 80 tiles per station).
```
Comment (AaronVanGeffen) [inline CHANGELOG.md:4]: Should change this.

File: src/OpenLoco/src/GameCommands/Track/CreateTrainStation.cpp [hunk 4]
```diff
@@ -42,14 +43,18 @@ namespace OpenLoco::GameCommands
         return result;
     }
 
-    static bool sub_48FEF4(StationId id, World::Pos3 pos)
+    // 0x0048FEF4
+    static bool exceedsStationLength(Station* station, World::Pos3 pos)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/GameCommands/Track/CreateTrainStation.cpp:47]: Used by other station game commands. Move to station manager.

**Comment by AaronVanGeffen** [comment]
This has been rebased to account for #2331. It's ready for review, as far as I'm concerned.

File: src/OpenLoco/src/World/StationManager.cpp [hunk 5]
```diff
@@ -527,6 +528,20 @@ namespace OpenLoco::StationManager
...
+    bool exceedsStationSize(Station& station, World::Pos3 pos)
+    {
+        if (Config::get().disableStationSizeLimit)
+            return false;
+
+        auto centreTile = World::Pos2(pos.x + World::kTileSize / 2, pos.y + World::kTileSize / 2);
+        auto stationCentre = World::Pos2(station.x, station.y);
+        auto distance = Math::Vector::manhattanDistance2D(centreTile, stationCentre);
+
+        const auto stationDistanceLimit = 8 * World::kTileSize;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/World/StationManager.cpp:541]: should make this a kStationDistanceLimit and move it out of the function.

Comment (AaronVanGeffen) [reply src/OpenLoco/src/World/StationManager.cpp:541]: What would be a good place for the constant? S5/Limits.h?

**Review by duncanspumpkin — APPROVED** [review]

File: src/OpenLoco/src/World/StationManager.cpp (hunk 5 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/World/StationManager.cpp:541]: hmm probably fine to just leave it in this file.

**Pull request merged**